src/nimlsp.nim

```diff
@@ -203,28 +210,31 @@ if not fileExists(nimpath / "config/nim.cfg"):
     "Supply the Nim project folder by adding it as an argument.\n"
   quit 1
 
-proc main(){.async.} =
+proc main(ins: Stream | AsyncFile, outs: Stream | AsyncFile) {.multisync.} =
   while true:
     try:
       debugLog "Trying to read frame"
-      let frame = await ins.readFrame
+      let frame = when ins is Stream: ins.readFrame else: await ins.readFrame
```

Comment:
Maybe use the `multisyncTask` template here as well?

Reply:
it seems it does not work for assignment, it throw error.

Reply:
If you make the `multisyncTask` template return `untyped` then it should work

Reply:
yeah, it works.